### PR TITLE
Uf 4537 refactor tags api

### PR DESCRIPTION
**Pull request by josc04**
- Removed tag suffix and renamed path from tags to metadata
- Added resources for creating, updating and deleting different metadata (implementation is done in separate tasks)

### CODE REVIEW

File: src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidStatusTagConstraintValidator.java [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
 import javax.validation.ConstraintValidatorContext;
 
 import org.springframework.beans.factory.annotation.Autowired;
 
+import se.sundsvall.supportmanagement.api.model.metadata.Status;
 import se.sundsvall.supportmanagement.api.validation.ValidStatusTag;
-import se.sundsvall.supportmanagement.service.TagService;
+import se.sundsvall.supportmanagement.service.MetadataService;
 
 public class ValidStatusTagConstraintValidator extends AbstractTagConstraintValidator implements ConstraintValidator<ValidStatusTag, String> {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidStatusTagConstraintValidator.java:17]: Remove tag-suffix here as well (both in the name of this class and the parent class)?

File: src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidTypeTagConstraintValidator.java [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
 import javax.validation.ConstraintValidatorContext;
 
 import org.springframework.beans.factory.annotation.Autowired;
 
+import se.sundsvall.supportmanagement.api.model.metadata.Type;
 import se.sundsvall.supportmanagement.api.validation.ValidTypeTag;
-import se.sundsvall.supportmanagement.service.TagService;
+import se.sundsvall.supportmanagement.service.MetadataService;
 
 public class ValidTypeTagConstraintValidator extends AbstractTagConstraintValidator implements ConstraintValidator<ValidTypeTag, String> {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidTypeTagConstraintValidator.java:17]: Remove tag-suffix?

File: src/main/java/se/sundsvall/supportmanagement/service/MetadataService.java [hunk 3]
```diff
@@ -34,50 +36,54 @@ public class TagService {
...
 			.toList();
 	}
 
-	@Cacheable(value = "tagCache", key = "{#root.methodName, #namespace, #municipalityId, #category}")
-	public List<String> findAllTypeTags(String namespace, String municipalityId, String category) {
+	@Cacheable(value = "metadataCache", key = "{#root.methodName, #namespace, #municipalityId, #category}")
+	public List<Type> findTypes(String namespace, String municipalityId, String category) {
 		return categoryRepository.findAllByNamespaceAndMunicipalityId(namespace, municipalityId)
 			.stream()
+			.filter(Objects::nonNull)
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/supportmanagement/service/MetadataService.java:77]: Is this necessary? Will the categoryRepository return null references?

File: src/test/java/se/sundsvall/supportmanagement/api/validation/impl/ValidStatusTagConstraintValidatorTest.java [hunk 4]
```diff
@@ -41,7 +42,7 @@ class ValidStatusTagConstraintValidatorTest {
 	private ConstraintViolationBuilder constraintViolationBuilderMock;
 
 	@Mock
-	private TagService tagServiceMock;
+	private MetadataService tagServiceMock;
```
Comment (MickeWallner) [inline src/test/java/se/sundsvall/supportmanagement/api/validation/impl/ValidStatusTagConstraintValidatorTest.java:45]: metadataServiceMock?

File: src/test/java/se/sundsvall/supportmanagement/api/validation/impl/ValidTypeTagConstraintValidatorTest.java [hunk 5]
```diff
@@ -37,7 +38,7 @@ class ValidTypeTagConstraintValidatorTest {
 	private ConstraintViolationBuilder constraintViolationBuilderMock;
 
 	@Mock
-	private TagService tagServiceMock;
+	private MetadataService tagServiceMock;
```
Comment (MickeWallner) [inline src/test/java/se/sundsvall/supportmanagement/api/validation/impl/ValidTypeTagConstraintValidatorTest.java:41]: metadataServiceMock?

File: src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidCategoryTagConstraintValidator.java [hunk 6]
```diff
@@ -1,21 +1,31 @@
...
 
+import se.sundsvall.supportmanagement.api.model.metadata.Category;
 import se.sundsvall.supportmanagement.api.validation.ValidCategoryTag;
-import se.sundsvall.supportmanagement.service.TagService;
+import se.sundsvall.supportmanagement.service.MetadataService;
 
 public class ValidCategoryTagConstraintValidator extends AbstractTagConstraintValidator implements ConstraintValidator<ValidCategoryTag, String> {
 	@Autowired
-	private TagService tagService;
+	private MetadataService tagService;
```
Comment (MickeWallner) [inline src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidCategoryTagConstraintValidator.java:19]: metadataService?

File: src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidTypeTagConstraintValidator.java [hunk 7]
```diff
@@ -1,22 +1,31 @@
...
 	@Autowired
-	private TagService tagService;
+	private MetadataService tagService;
```
Comment (MickeWallner) [inline src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidTypeTagConstraintValidator.java:19]: metadataService?

File: src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidStatusTagConstraintValidator.java (hunk 1 above)
Comment (josc04) [reply src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidStatusTagConstraintValidator.java:17]: I'll leave it to Micke to change name as I will f*ck it up for him if I do the refactoring in my branch when he refactors the validators in his... I'll be sure to tell him in his PR if you don't 🙂!

File: src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidTypeTagConstraintValidator.java (hunk 2 above)
Comment (josc04) [reply src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidTypeTagConstraintValidator.java:17]: Same comment as in ValidStatusTagConstraintValidator

**Comment by josc04** [comment]
Acted/responded to all PR comments

**Review by MickeWallner — DISMISSED** [review]

**Review by MickeWallner — APPROVED** [review]

**Pull request merged**